src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx

```diff
@@ -74,16 +75,33 @@ const JazzbarEntranceExperience: React.FunctionComponent<PropsType> = ({
     purchaseHistoryRequestStatus: boolean;
   };
 
+  const futureOrOngoingVenueEvents =
+    venueEvents &&
+    venueEvents.filter(
+      (event: VenueEvent) =>
+        event.start_utc_seconds +
+          event.duration_minutes * ONE_MINUTE_IN_SECONDS >
+        Date.now() / 1000
+    );
+
   const { eventId, redirectTo } = getQueryParameters(location.search);
 
   venue && updateTheme(venue);
 
   useEffect(() => {
-    if (user && venueEvents && venueId && eventId && redirectTo === "payment") {
-      setSelectedEvent(venueEvents.find((event) => event.id === eventId));
+    if (
+      user &&
+      futureOrOngoingVenueEvents &&
+      venueId &&
+      eventId &&
+      redirectTo === "payment"
```

Comment:
this is quite hard to parse. Could you create a variable with a nice name e.g. const eventRequiringPayment = ...

Reply:
this code isn't usefule anymore as we use modals for the authentication
Thanks!